python_backend/services/user_preferences.py

```
@@ -1,8 +1,370 @@
+class PreferenceExtractor:
+    """사용자 프로필에서 선호도 추출 전담 클래스"""
+    
+    @staticmethod
+    def extract_from_profile(user_profile: Optional[Dict[str, Any]]) -> NegativePreferences:
+        """사용자 프로필로부터 네거티브 선호도 추출"""
+        if not user_profile:
+            return PreferenceExtractor._get_default_preferences()
+        
+        user_id = user_profile.get('userId', 'unknown')
+        formality = user_profile.get('baseFormalityLevel', 3)
+        friendliness = user_profile.get('baseFriendlinessLevel', 3)
```

Ruff (0.12.2)

Local variable `friendliness` is assigned to but never used

Remove assignment to unused variable `friendliness`

(F841)